src/post.rs

```diff
@@ -140,48 +211,272 @@ impl Post {
     }
 }
 
-#[allow(clippy::module_name_repetitions)]
-#[derive(Serialize)]
-#[serde(rename_all = "camelCase")]
-struct ApiPost<'a> {
-    adult_content: bool,
-    blocks: Vec<ApiBlock<'a>>,
-    cws: &'a [String],
-    headline: &'a str,
-    post_state: u64,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    share_of_post_id: Option<PostId>,
-    tags: &'a [String],
+impl From<de::Post> for Post {
+    fn from(api: de::Post) -> Self {
+        let locations = PostLocations {
+            id: api.post_id,
+            filename: api.filename,
+            url: api.single_post_page_url,
+            edit_url: api.post_edit_url,
+        };
+        let metadata = PostMetadata {
+            locations,
+            can_share: api.can_share,
+            comments_locked: api.comments_locked,
+            has_any_contributor_muted: api.has_any_contributor_muted,
+            has_cohost_plus: api.has_cohost_plus,
+            liked: api.is_liked,
+            num_comments: api.num_comments,
+            num_shared_comments: api.num_shared_comments,
+            pinned: api.pinned,
+            related_projects: {
+                let mut related_projects: Vec<String> = api
+                    .related_projects
+                    .into_iter()
+                    .map(|project| project.handle)
+                    .collect();
+                if related_projects.is_empty() {
+                    related_projects.push(api.posting_project.handle.clone());
```

Comment:
The Cohost API doesn't include the poster's handle unless there is something in the share tree. For consistency I push it in here if it's missing.